iroh-ctl/src/p2p.rs

```diff
@@ -0,0 +1,306 @@
+    command: GossipsubCommands,
+}
+
+#[derive(Subcommand, Debug, Clone)]
+pub enum GossipsubCommands {
+    Publish {
+        topic: String,
+        #[clap(long, short)]
+        file: PathBuf,
+    },
+    Subscribe {
+        topic: String,
+    },
+    Unsubscribe {
+        topic: String,
+    },
+}
+
+pub async fn run_command(rpc: Client, cmd: P2p) -> Result<()> {
+    match cmd.command {
+        P2pCommands::Version => {
+            let v = rpc.p2p.version().await?;
+            println!("v{}", v);
+        }
+        P2pCommands::Addrs(addrs) => match addrs.command {
+            None => {
+                let addrs = rpc.p2p.get_peers().await?;
+                println!("{:#?}", addrs);
+            }
+            Some(AddrsCommands::Listen) => {
+                let addrs = rpc.p2p.get_listening_addrs().await?;
+                println!("{:#?}", addrs);
+            }
+            Some(AddrsCommands::Local) => {
+                todo!("Local not yet implemented.");
+            }
+        },
+        P2pCommands::Connect { peer_id, addrs } => {
+            rpc.p2p.connect(peer_id, addrs).await?;
+            println!("connected to {}", peer_id);
+        }
+        P2pCommands::Disconnect { peer_id } => {
+            rpc.p2p.disconnect(peer_id).await?;
+            println!("disconnected from {}", peer_id);
+        }
+        P2pCommands::Peers => {
+            let peers: Vec<PeerId> = rpc.p2p.get_peers().await?.into_keys().collect();
+            println!("{:#?}", peers);
+        }
+        P2pCommands::Ping { ping_args, count } => {
+            todo!("{:?} {:?}", ping_args, count);
+        }
+        P2pCommands::Dht(d) => {
+            todo!("DHT commands are not yet implemented - {:#?}", d);
+        }
+        P2pCommands::Bitswap(d) => {
+            todo!("Bitswap commands are not yet implemented - {:#?}", d);
+        }
+        P2pCommands::Dev(dev) => match dev.command {
+            DevCommands::FetchBitswap { cid, providers } => {
+                let providers = HashSet::from_iter(providers.into_iter());
+                let res = rpc.p2p.fetch_bitswap(cid, providers).await?;
+                println!("{:#?}", res);
+            }
+            DevCommands::FetchProviders { cid } => {
+                let res = rpc.p2p.fetch_providers(&cid).await?;
+                println!("{:#?}", res);
+            }
+            DevCommands::Gossipsub(g) => match g.command {
+                GossipsubCommands::Publish { topic, file } => {
+                    let mut f = std::fs::File::open(file)?;
+                    let mut v: Vec<u8> = Vec::new();
+                    f.read_to_end(&mut v)?;
+                    let message_id = rpc
+                        .p2p
+                        .gossipsub_publish(TopicHash::from_raw(topic), Bytes::from(v))
+                        .await?;
+                    println!("Message Id: {}", message_id);
+                }
+                GossipsubCommands::Subscribe { topic } => {
```

Comment:
How do I see incoming messages on gossipsub? are they printed if I am subscribed? maybe add a command to tail a topic?